src/revivalpmmp/pureentities/event/EventListener.php

```diff
@@ -76,7 +76,7 @@ public function dataPacketReceiveEvent(DataPacketReceiveEvent $event){
 		if($btnTxt === null){
 			return;
 		}
-		if($packet->transactionType !== InventoryTransactionPacket::TYPE_USE_ITEM_ON_ENTITY){
+		if($packet->trData !== InventoryTransactionPacket::TYPE_USE_ITEM_ON_ENTITY){
```

Comment:
Here you should use `$packet->trData instanceof UseItemOnEntityTransactionData` instead

Reply:
wtf... no

Reply:
Look here https://github.com/pmmp/PocketMine-MP/blob/c9b83d72761b36034f5c2d2e8c1668ba6c55c08c/src/pocketmine/network/mcpe/protocol/InventoryTransactionPacket.php#L78

Reply:
Please read the changes carefully. He is right. You should use the 
`!$packet->trData instanceof UseItemOnEntityTransactionData`

Reply:
it works, that's the most important thing

Reply:
The most important thing is that the code is written correctly.

Reply:
I made the requested changes for you, just review and merge my pull request into your branch https://github.com/supercrafter333/PureEntitiesX/pull/1